src/main/scala/org/renci/cam/LookupService.scala

```diff
@@ -0,0 +1,420 @@
+        val predResults = results map { res =>
+          val predIRI = res.getResource("p").getURI
+
+          val pred = res.getLiteral("pLabel") match {
+            case null => LabeledIRI(predIRI, Set())
+            case lit  => LabeledIRI(predIRI, Set(lit.getString))
+          }
+
+          val biolinkRes = biolinkRelationMap.get(IRI(predIRI))
+          val biolink = biolinkRes match {
+            case None                     => Set[LabeledIRI]()
+            case Some((None, iri))        => Set(LabeledIRI(iri.value, Set()))
+            case Some((Some(label), iri)) => Set(LabeledIRI(iri.value, Set(label)))
+          }
+
+          (pred, biolink)
+        }
+
+        val objLabeled = results
+          .map(
+            { res =>
+              res.getLiteral("objLabel") match {
+                case null => LabeledIRI(res.getResource("obj").getURI, Set())
+                case lit  => LabeledIRI(res.getResource("obj").getURI, Set(lit.getString))
+              }
+            }
+          )
+          .toSet
+
+        val subjLabeled = results
+          .map(
+            { res =>
+              res.getLiteral("subjLabel") match {
+                case null => LabeledIRI(res.getResource("subj").getURI, Set())
+                case lit  => LabeledIRI(res.getResource("subj").getURI, Set(lit.getString))
+              }
+            }
+          )
+          .toSet
+
+        Relation(
+          subjLabeled,
+          predResults.map(_._1).toSet,
+          predResults.map(kv => (kv._1.iri, kv._2)).toMap,
+          objLabeled,
+          Seq[Relation](), // TODO: recurse!
+          results.map(_.getResource("g").getURI).toSet
+        )
+      }
+    } yield relations
+
+  /* Node Norm structures and functions */
+
+  /* Node Norm result structures */
+
+  /** Node Norm identifier */
+  case class NodeNormIdentifier(
+    identifier: String,
+    label: Option[String]
+  ) {
+    val asLabeledIRI = LabeledIRI(identifier, label.toSet)
+  }
+
+  /** Node Norm response */
+  case class NodeNormResponse(
+    id: Option[NodeNormIdentifier],
+    equivalent_identifiers: List[NodeNormIdentifier],
+    `type`: List[BiolinkClass],
+    information_content: Option[Float]
+  )
+
+  /** Retrieve a list of all equivalent identifiers for a queryId.
+    *
+    * @param queryId
+    *   The identifier being queried.
+    * @param nodeNormURL
+    *   The NodeNorm URL (ending with `/get_normalized_nodes`) to query.
+    * @param conflate
+    *   What to set in the `conflate` param to NodeNorm. Currently, may be `true` or `false`.
+    */
+  def getQualifiedIdsFromNodeNorm(queryId: String,
+                                  nodeNormURL: String,
+                                  conflate: String): ZIO[Blocking with Has[BiolinkData], Exception, List[LabeledIRI]] =
+    for {
+      biolinkData <- biolinkData
+      nnUri <- ZIO.fromEither(Uri.fromString(nodeNormURL))
+      uri = nnUri.+?("curie", queryId).+?("conflate", conflate).toString()
```

Comment:
What is `+?`?

Reply:
Replaced with the easier to read `.withQueryParam()` in 250c718.